Is there something I can tell you about my cyruslibs install that will confirm whether it's old enough?  Oh actually (happening as I type), here's a thing -- ical_support no longer building for me:

```
imap/ical_support.c: In function 'multirrule_iterator_add':
imap/ical_support.c:519:5: error: implicit declaration of function 'icalrecurrencetype_ref'; did you mean 'icalrecurrencetype_unref'? [-Werror=implicit-function-declaration]
  519 |     icalrecurrencetype_ref(recur);
      |     ^~~~~~~~~~~~~~~~~~~~~~
      |     icalrecurrencetype_unref
  CC       imap/libcyrus_imap_la-webdav_db.lo
cc1: all warnings being treated as errors
```

--- a/imap/ical_support.c
+++ b/imap/ical_support.c
@@ -516,7 +516,11 @@ static void multirrule_iterator_add(struct multirrule_iterator *iter,
     entry->icaliter = icaliter;
     entry->next = icalrecur_iterator_next(entry->icaliter);
     entry->recur = recur;
+#ifdef HAVE_RECUR_BY_REF
+    icalrecurrencetype_ref(recur);
+#else /* !HAVE_RECUR_BY_REF */
     entry->recur->rscale = xstrdupnull(recur->rscale);
+#endif /* HAVE_RECUR_BY_REF */
 }
 
 static icaltimetype multirrule_iterator_next(struct multirrule_iterator *iter)
@@ -574,9 +578,14 @@ multirrule_iterator_for_range(icalcomponent *comp,
         if (!span_compare_range(&recur_span, &range_span)) {
             multirrule_iterator_add(&iter, recur, dtstart, range.start);
         }
+
+#ifdef HAVE_RECUR_BY_REF
+        icalrecurrencetype_unref(recur);
+#else /* !HAVE_RECUR_BY_REF */
         else {
             icalrecurrencetype_unref(recur);
         }
+#endif
     }
 
     return iter;

